pymmmc/tests/clv/models/test_beta_geo.py

```diff
@@ -0,0 +1,205 @@
+
+
+class TestBetaGeoModel:
+    @classmethod
+    def setup_class(cls):
+        rng = np.random.default_rng(34)
+
+        # Hyperparameters
+        a_true = 0.8
+        b_true = 2.5
+        alpha_true = 3
+        r_true = 4
+
+        # Number of subjects
+        N = 500
+        # Subject level parameters
+        p = pm.draw(pm.Beta.dist(a_true, b_true, size=N), random_seed=rng)
+        lam = pm.draw(pm.Gamma.dist(r_true, alpha_true, size=N), random_seed=rng)
+
+        T = pm.draw(
+            pm.DiscreteUniform.dist(lower=20, upper=40, size=N), random_seed=rng
+        )
+
+        data = continuous_contractual.rng_fn(rng, lam, p, T, 0, size=N)
+
+        recency = data[..., 0]
+        frequency = data[..., 1]
+        alive = 1 - data[..., 2]
+
+        cls.rng = rng
+        cls.a_true = a_true
+        cls.b_true = b_true
+        cls.alpha_true = alpha_true
+        cls.r_true = r_true
+        cls.N = N
+        cls.recency = recency
+        cls.frequency = frequency
+        cls.T = T
+        cls.alive = alive
+
+        cls.fixed_model = BetaGeoModel(
+            frequency=cls.frequency,
+            recency=cls.recency,
+            T=cls.T,
+        )
+        cls.fixed_model.fit(chains=1, progressbar=False, random_seed=cls.rng)
+
+    @pytest.mark.parametrize("a_prior", (None, pm.HalfNormal.dist()))
+    @pytest.mark.parametrize("b_prior", (None, pm.HalfStudentT.dist(nu=4)))
+    @pytest.mark.parametrize("alpha_prior", (None, pm.HalfCauchy.dist(2)))
+    @pytest.mark.parametrize("r_prior", (None, pm.Gamma.dist(1, 1)))
+    def test_model(self, a_prior, b_prior, alpha_prior, r_prior):
+        model = BetaGeoModel(
+            frequency=self.frequency,
+            recency=self.recency,
+            T=self.T,
+            a_prior=a_prior,
+            b_prior=b_prior,
+            alpha_prior=alpha_prior,
+            r_prior=r_prior,
+        )
+
+        assert isinstance(
+            model.model["a"].owner.op,
+            pm.HalfFlat if a_prior is None else type(a_prior.owner.op),
+        )
+        assert isinstance(
+            model.model["b"].owner.op,
+            pm.HalfFlat if b_prior is None else type(b_prior.owner.op),
+        )
+        assert isinstance(
+            model.model["alpha"].owner.op,
+            pm.HalfFlat if alpha_prior is None else type(alpha_prior.owner.op),
+        )
+        assert isinstance(
+            model.model["r"].owner.op,
+            pm.HalfFlat if r_prior is None else type(r_prior.owner.op),
+        )
+        assert model.model.eval_rv_shapes() == {
+            "a": (),
+            "a_log__": (),
+            "b": (),
+            "b_log__": (),
+            "alpha": (),
+            "alpha_log__": (),
+            "r": (),
+            "r_log__": (),
+        }
+
+    def test_model_convergence(self):
+        # b parameter has the largest mismatch of the four parameters
+        # rtol=0.3 should be enough
```

Comment:
Can you confirm that if you increase your data size the mismatch gets smaller?

Reply:
I am able to decrease `rtol` with increasing `N`

Reply:
Good enough, no need to change it

Reply:
I already added it, it can be easily removed. Hopefully it doesn't add too much computation to our tests. I'll force push the change this afternoon.